### PR TITLE
Some codes are duplicated and optimized

### PR DESCRIPTION
**Pull request by hellomyboy**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

I found some duplicate code, and redesigned and optimized it.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)no
  - The public API: (yes / no)no
  - The schema: (yes / no / don't know)no
  - The default values of configurations: (yes / no)no
  - The wire protocol: (yes / no)no
  - The rest endpoints: (yes / no)no
  - The admin cli options: (yes / no)no
  - Anything that affects deployment: (yes / no / don't know)no

### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 

- [ ] doc-required 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] no-need-doc 
  
  (Please explain why)
  
- [ ] doc 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java
Comment (michaeljmarshall) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java:286]: I think this method needs a better name to indicate what it is actually doing. It's probably sufficient to overload the `asyncGetRequest` method name here and make its signature: 

```java
public <T> CompletableFuture<T> getCompletableFuture(WebTarget path)
```

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
Good find.

@codelipenghui, since you wrote #9203, would you like to take a look?

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java
Comment (hellomyboy) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java]: @michaeljmarshall hello，Why not reply to me, I have been waiting for you to reply to the message, but I have not answered, I am very anxious, the optimization does not conform to the open source standards, if there is not meet the norms, please make suggestions, I hope to be able to participate.

**Review by michaeljmarshall — DISMISSED** [review]
LGTM.

@hellomyboy - I am sorry for my delayed review. When I first reviewed this, I wasn't a committer, so I couldn't merge it and my review had less weight. Now, I am a committer, so I can merge this.

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java
Comment (michaeljmarshall) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java]: Looks good to me. Sorry, again, for my delayed review.

**Comment by michaeljmarshall** [comment]
@hellomyboy - are you able to resolve conflicts? Then we can run tests and merge this PR.

**Review by codelipenghui — DISMISSED** [review]

**Review by Jason918 — APPROVED** [review]

**Comment by codelipenghui** [comment]
@hellomyboy Could you please help check the failed tests?

**Review by Shoothzj — APPROVED** [review]

**Review by michaeljmarshall — APPROVED** [review]

**Comment by codelipenghui** [comment]
@hellomyboy Looks like the failed test is related to this PR, I have created a flaky test https://github.com/apache/pulsar/issues/14699 issue, but looks like it happens in 2 tests of this PR, looks like not a flaky test.

**Comment by tisonkun** [comment]
Closed as stale. Feel free to rebase and resubmit the patch.

**Pull request closed without merging**
